blocks/algolia-assortment-content-source-block/sources/algolia-assortment.js

```diff
@@ -0,0 +1,49 @@
+import algoliasearch from "algoliasearch/lite";
+import { algoliaSearchKey, algoliaAppId } from "fusion:environment";
+
+const params = {
+	filters: "text",
+	index: "text",
+	query: "text",
+	ruleContexts: "text",
+	hitsPerPage: "number",
+	page: "number",
+};
+
+const fetch = async (contentSourceParameters) => {
+	const { filters, index, query, ruleContexts } = contentSourceParameters;
+	let { hitsPerPage, page } = contentSourceParameters;
+
+	// parameters in content source are defined as null by default
+	if (hitsPerPage === null) {
+		hitsPerPage = 20;
+	}
+	if (page === null) {
+		page = 0;
+	}
+	const client = algoliasearch(algoliaAppId, algoliaSearchKey);
+
+	const clientSearchIndex = client.initIndex(index);
+	try {
+		const { hits } = await clientSearchIndex.search(query, {
+			// array of strings
+			// https://www.algolia.com/doc/api-reference/api-parameters/ruleContexts/
+			ruleContexts: [ruleContexts],
+			filters,
+			hitsPerPage,
+			page,
+		});
```

Comment:
Also for another day, but when the Commerce schema is final we can use this as a filter to minimize payload:
https://www.algolia.com/doc/api-reference/api-parameters/attributesToRetrieve/

Reply:
Yep good point. Is this ok to resolve?